lib/core/src/system/gpu/data/texture.rs

```diff
@@ -326,49 +331,72 @@ with_texture_format_relations!(generate_internal_format_instances []);
 
 
 
-// =====================
-// === TextureSource ===
-// =====================
+// =======================
+// === Texture sources ===
+// =======================
+
+// === Empty Texture ===
+
+/// An empty texture, created as 1x1 texture in Rgba format.
+pub struct EmptyTexture;
+
 
-/// Source of the texture. Please note that the texture will be loaded asynchronously on demand.
-#[derive(Clone,Debug)]
-pub enum TextureSource {
-    /// URL the texture should be loaded from. This source implies asynchronous loading.
-    Url(String)
+
+// === Texture from url ===
+
+/// Texture loaded from given url. It is loaded to gpu asynchronously, putting a mocked texture
+/// before
+pub struct TextureFromUrl {
+    /// An url from where the texture is loaded.
+    pub url       : String,
 }
```

Comment:
This should be named `Url` instead. This type is very generic, it does not have anything in common with textures.